docs/MOTD.md

```diff
@@ -0,0 +1,362 @@
+
+#### Validation Criteria
+
+1. **Packet Size:**  
+   The total size must be exactly 33 bytes.
+2. **Packet ID:**  
+   Must strictly be `0x01`.
+3. **Timestamp:**  
+   It is just a timestamp of uptime, but we can check it is not infront of unix timestamp
+4. **Magic Offline Message ID:**  
+   Must match the predefined 16-byte sequence.
+5. **GUID:**  
+   It doesn't really need to be checked, but it is there
+
+### Unconnected Pong
+
+In response to a valid unconnected ping, the server dispatches an unconnected pong packet containing the MOTD and additional server details.
+
+- **Packet ID (1 byte):** `0x1C` (Denotes Unconnected Pong)
+- **Timestamp (8 bytes, Big-Endian):** The time since start of the server. Just setting it to current unix timestamp works fine
+- **Server GUID (8 bytes, Big-Endian):** Unique server identifier.
+- **Magic Offline Message ID (16 bytes):** Identical to the one used in the ping.
+- **MOTD Length (2 bytes, Big-Endian):** Specifies the length of the MOTD string.
+- **MOTD (Variable Length):** The actual Message of the Day in byte format.
+
+## Message of the Day (MOTD) Generation
+
+The MOTD is what shows up in the server list, it is what people see before joining the server.
+
+### MOTD Format
+
+The MOTD follows a semicolon-separated (`;`) format comprising various fields, semi colons can be escaped with a backslash (`\`)
+```
+<Edition>;<MOTD>;<Protocol Version>;<Game Version>;<Player Count>;<Max Player Count>;<Server GUID>;<World Name>;<Game Mode>;<Nintendo Limited>;<IpV4Port>;<IpV6Port>;<Extra Data>;
+```
+
+#### Field Breakdown & Data Types
+
+- **Edition:** string  
+  Specifies the game edition (e.g., `MCPE` for Minecraft Pocket Edition), seems to be hardcoded to `MCPE`
+- **MOTD:** string  
+  The actual message that is displayed, it can have Minecraft colors.  
+- **Protocol Version:** uint16  
+  Numeric identifier indicating the game protocol version (e.g., `748`).  
+- **Game Version:** string  
+  Denotes the server's game version (e.g., `1.20.40`, or on CubeCraft: `1`).  
+- **players online:** uint32  
+  Current number of active players (e.g., `10`).  
+- **players max:** uint32  
+  Maximum number of players allowed on the server (e.g., `100`).  
+- **server id:** uint64  
+  Unique identifier for the server (the server GUID but as a number).  
+- **level name:** string  
+  Name of the game world (e.g., `world`, some docs call this submotd).  
+- **gamemode:** string  
+  Current game mode (e.g., `Survival`, `Creative`, `Adventure`).  
+- **NintendoLimited:** boolean  
+  Returns the status of Nintendo's limitation to join (0 or 1).  
+- **port v4:** uint16  
+  The port of the server for IPv4, useful for servers that have both IPv4 and IPv6.  
+- **port v6:** uint16  
+  The port of the server for IPv6, useful for servers that have both IPv4 and IPv6.  
+- **Extra Data:**  
+  Ends with `0` to signify the conclusion of the MOTD string.  
+
+## Code Examples
+
+::: code-group
```

Comment:
wonder if there is a way to make this collapsible, since not everyone might want to see it directly..